sdk/apis/apis/v1alpha1/types_apiexportendpointslice.go

```diff
@@ -73,10 +73,19 @@ type APIExportEndpointSliceStatus struct {
 	// conditions is a list of conditions that apply to the APIExportEndpointSlice.
 	Conditions conditionsv1alpha1.Conditions `json:"conditions,omitempty"`
 
+	// endpoints contains all the URLs of the APIExport service.
+	//
 	// +optional
+	// +listType=map
+	// +listMapKey=url
+	APIExportEndpoints []APIExportEndpoint `json:"endpoints"`
 
-	// endpoints contains all the URLs of the APIExport service.
-	APIExportEndpoints []APIExportEndpoint `json:"endpoints,omitempty"`
+	// +optional
+
+	// shardSelector is the selector used to filter the shards. It is used to filter the shards
+	// when determining partition scope when deriving the endpoints. This is set by owning shard,
+	// and is used by follower shards to determine if its inscope or not.
+	ShardSelector string `json:"shardSelector,omitempty"`
```

Comment:
This is needed so we can cleanly split reconcilers. One uses committer, another - SSA client. Keeping them together gets very much non-readable code.